### ISSUE TITLE
Add curryable zip2/zip3/etc, combine2/combine3/etc?

### ISSUE THREAD

**Issue by briancavalier** [mostjs/core#10]
One way to solve the "problem" of variadic functions not being curry-friendly is to provide variants with specific arities.  Haskell does this, i.e. zip2, zip3, et al.  We could do the same by keeping the variadic versions and simply providing fixed-arity wrappers.  For example:

```js
import { combine } from './combinators/combine'

export { combine } // variadic

export combine2 = curry3((f, s1, s2) => combine(f, s1, s2))
export combine3 = // ...
// etc
```

Of course, there's always the option of waiting until folks ask for this before actually implementing it.  But, I wanted to get some current thoughts on it.

**Comment by TylorS** [comment]
I think it'd be easier to have `combine(f, s1, s2)` as a non-variadic function, but then to simply have `combineArray(f, [ ...streams ])`. A more radical approach would to remove `combine` altogether in favor of `combineArray`, which then, might be able to simple be `combine(f, [ ...streams])`.

**Comment by davidchase** [comment]
I think i kinda like the `combine(f, s1, s2)` and also the `combineArray(f, [... streams])`, though i am guessing knowing the arity (`zip2, zip3`) ahead of time is better for the program? so something like `combineN(2, f, s1, s2)` would not be ideal ?

**Comment by jethrolarson** [comment]
I don't mind `combine2` etc. 
With ramda I think I'd rather have curry3 than curryN(3), though that means they all have to be documented and tested which may have diminishing returns. 

BTW is combine2 basically the same as liftA2?

**Comment by briancavalier** [comment]
> is combine2 basically the same as liftA2?

Yep.  Variadic `combine` is basically variadic `liftA`

> they all have to be documented and tested which may have diminishing returns

Good point.  I think you're right about diminishing returns ... combine4 is probably more rare in the wild than combine3, and so on.  Going with 2 and 3 is probably enough for the vast majority cases.  Since they'd all be implemented using variadic `combine`, the additional tests could be pretty minimal.  I'll bet docs could be similarly minimal: "A fixed argument version of combine that allows partial application" or somesuch might be close to enough.

The thing that variadic `combine/zip` *and* array-based `combineArray/zipArray` each lack is partial application, i.e. I can't fix the function and the first stream and demand the second, or fix only the function to lift a 2-argument function *which is also partially applicable*. without using a partial application helper (or `bind`).  Partial application aside, I don't think there's any benefit to providing fixed-arity versions.

Perhaps @TylorS's suggestion is the way to start?  Make `combine` fixed-arity, and let `combineArray` serve as the "variadic" version for `@most/core`?

**Comment by davidchase** [comment]
closed by #68 thanks @joshburgess
